### PR TITLE
Add app deploy to init

### PR DESCRIPTION
**Pull request by sonofachamp**
Sample output:
```
→ ./bin/local/archer init
Note: It's best to run this command in the root of your workspace.
Welcome to the ECS CLI! We're going to walk you through some questions 
to help you get set up with a project on ECS. A project is a collection of 
containerized applications (or micro-services) that operate together.

Looks like you don't have any existing projects. Let's create one!
? What would you like to call your project? myproj
? Which type of infrastructure pattern best represents your application? Load Balanced Web App
? What do you want to call this Load Balanced Web App? myapp
? Which Dockerfile would you like to use for myapp app? myapp/Dockerfile
Ok great, we'll set up a Load Balanced Web App named myapp in project myproj.
✔ Success! Created the infrastructure to manage container repositories under project myproj.

✔ Success! Wrote the manifest for myapp app at 'ecs-project/myapp-app.yml'
Your manifest contains configurations like your container size and ports.

✔ Success! Created ECR repositories for application myapp.
2019/11/18 15:58:38 Created Application with version 1
All right, you're all set for local development.
? Would you like to deploy a staging environment? Yes
✔ Success! Created the infrastructure for the test environment.
- Virtual private cloud on 2 availability zones to hold your services     [Complete]
  - Internet gateway to connect the network to the internet               [Complete]
  - Public subnets for internet facing services                           [Complete]
  - Private subnets for services that can't be reached from the internet  [Complete]
  - NAT gateway for private services to send requests to the internet     [Complete]
  - Routing tables for services to talk with each other                   [Complete]
- ECS Cluster to hold your services                                       [Complete]
- Application load balancer to distribute traffic                         [Complete]
✔ Success! Linked account <accountID> and region us-west-2 project myproj.
2019/11/18 16:04:15 Created environment with version 1
✔ Success! Created environment test in region us-west-2 under project myproj.
Only found one app, defaulting to: myapp
Sending build context to Docker daemon  57.34kB
Step 1/2 : FROM nginx
 ---> e445ab08b2be
Step 2/2 : COPY index.html /usr/share/nginx/html
 ---> Using cache
 ---> 711d8eb7377b
Successfully built 711d8eb7377b
Successfully tagged <accountID>.dkr.ecr.us-west-2.amazonaws.com/myproj/myapp:05b1631
WARNING! Your password will be stored unencrypted in <path>/.docker/config.json.
Configure a credential helper to remove this warning. See
https://docs.docker.com/engine/reference/commandline/login/#credentials-store

Login Succeeded
The push refers to repository [<accountID>.dkr.ecr.us-west-2.amazonaws.com/myproj/myapp]
c2c3319ed52b: Pushed 
fe6a7a3b3f27: Pushed 
d0673244f7d4: Pushed 
d8a33133e477: Pushed 
05b1631: digest: sha256:86ebd369638102b284c87cf8837d9f8b2241d5c8d467ea4589beff7fd5ebde55 size: 1155
Done!
✔ Success! Deployed myapp:05b1631 to test.
```

**NOTE:** I was fighting `gomock` to generate a mock `appDeployer` and it basically wants everything to be exported which seems unnecessary, so for now I opted to create a simple inline `mockAppDeployer` that makes testing the happy path scenario easy.

#297 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/cli/init.go [hunk 1]
```diff
@@ -153,8 +171,8 @@ func (opts *InitOpts) loadApp() error {
 	return opts.initApp.Validate()
 }
 
-// deploy prompts the user to deploy a test environment if the project doesn't already have one.
-func (opts *InitOpts) deploy() error {
+// deployEnv prompts the user to deployEnv a test environment if the project doesn't already have one.
```
Comment (SoManyHs) [inline internal/pkg/cli/init.go]: Nit: second "deploy" shouldn't be changed to "deployEnv" right?

File: internal/pkg/cli/init_test.go [hunk 2]
```diff
@@ -12,6 +12,20 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+type mockAppDeployer struct{}
+
+func (mad mockAppDeployer) init() error {
```
Comment (SoManyHs) [inline internal/pkg/cli/init_test.go]: lol `mad`

Comment (SoManyHs) [reply internal/pkg/cli/init_test.go]: tell us how you really feel @sonofachamp

File: internal/pkg/cli/init.go (hunk 1 above)
Comment (sonofachamp) [reply internal/pkg/cli/init.go]: Ah yes, the woes of replace all.

File: internal/pkg/cli/init_test.go (hunk 2 above)
Comment (sonofachamp) [reply internal/pkg/cli/init_test.go]: It may or may not have been related to fighting `gomock`

**Pull request merged**